From ec8cac42a6581d21f34001ec9a04385ecd1a5838 Mon Sep 17 00:00:00 2001
From: Cyril Arnould <cyril.arnould@outlook.com>
Date: Wed, 14 Jun 2023 00:11:24 +0200
Subject: [PATCH 14/16] Require signals to be defined

---
 find/exec.c   | 2 ++
 xargs/xargs.c | 4 +++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/find/exec.c b/find/exec.c
index 3d0a8c51..ab8bfe96 100644
--- a/find/exec.c
+++ b/find/exec.c
@@ -392,7 +392,9 @@ launch (struct buildcmd_control *ctl, void *usercontext, int argc, char **argv)
   if (first_time)
     {
       first_time = 0;
+#ifdef SIGCHLD
       signal (SIGCHLD, SIG_DFL);
+#endif
     }
   
 #if defined(__MINGW32__) || defined(_MSC_VER) || defined(__MSDOS__)
diff --git a/xargs/xargs.c b/xargs/xargs.c
index 53ba33df..a1e882e8 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -839,8 +839,10 @@ main (int argc, char **argv)
   bc_state.argbuf = xmalloc (bc_ctl.arg_max + 1);
 
   /* Make sure to listen for the kids.  */
+#ifdef SIGCHLD
   signal (SIGCHLD, SIG_DFL);
-
+#endif
+  
   if (!bc_ctl.replace_pat)
     {
       for (; optind < argc; optind++)
-- 
2.45.2

